hopper/src/main/java/org/atomhopper/abdera/parser/UnifiedParser.java

```diff
@@ -0,0 +1,389 @@
+package org.atomhopper.abdera.parser;
+
+import org.apache.abdera.model.Document;
+import org.apache.abdera.model.Element;
+import org.apache.abdera.parser.ParseException;
+import org.apache.abdera.parser.Parser;
+import org.apache.abdera.parser.ParserOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import javax.xml.stream.XMLStreamReader;
+import java.io.InputStream;
+import java.io.Reader;
+import java.nio.channels.ReadableByteChannel;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+@Component
+public class UnifiedParser implements Parser {
+
+    private final Parser xmlParser;
+    private final JsonAtomParser jsonParser;
+    private ParserOptions defaultParserOptions;
+
+    // Supported JSON content types
+    private static final Set<String> SUPPORTED_JSON_TYPES = new HashSet<>(Arrays.asList(
+            "application/json",
+            "application/atom+json",
+            "text/json"
+    ));
+
+    // Supported XML content types
+    private static final Set<String> SUPPORTED_XML_TYPES = new HashSet<>(Arrays.asList(
+            "application/xml",
+            "application/atom+xml",
+            "text/xml"
+    ));
+
+    static Logger LOG = LoggerFactory.getLogger(UnifiedParser.class);
+
+    public UnifiedParser(Parser xmlParser) {
+        if (xmlParser == null) {
+            LOG.error("XML parser is null in constructor");
+            throw new IllegalArgumentException("XML parser cannot be null");
+        }
+        this.xmlParser = xmlParser;
+        this.jsonParser = new JsonAtomParser();
+        this.defaultParserOptions = xmlParser.getDefaultParserOptions();
+        LOG.info("UnifiedParser is initialised with XMLParser: {}", xmlParser.getClass().getSimpleName());
+    }
+
+
+    private boolean isJsonContent(String contentType) throws ParseException {
+        validateContentType(contentType);
+        String normalized = normalizeContentType(contentType);
+        return SUPPORTED_JSON_TYPES.contains(normalized);
+    }
+
+    private boolean isXmlContent(String contentType) throws ParseException {
+        validateContentType(contentType);
+        String normalized = normalizeContentType(contentType);
+        return SUPPORTED_XML_TYPES.contains(normalized);
+    }
+
+    private void validateContentType(String contentType) throws ParseException {
+        if (contentType == null || contentType.trim().isEmpty()) {
+            LOG.error("Content-Type header is missing or empty");
+            throw new ParseException("Content-Type header is required and cannot be null or empty");
+        }
+    }
+
+    private String normalizeContentType(String contentType) {
+        if (contentType == null) {
+            return null;
+        }
+
+        String normalized = contentType.toLowerCase().split(";")[0].trim();
+        return normalized;
+    }
```

Comment:
Return the function result instead of creating the `normalized` variable. You already escape hatched the null case so the String is instantiated and you know the modifications will complete.